sphinx/domains/python.py

```diff
@@ -91,7 +91,14 @@ def type_to_xref(text: str, env: BuildEnvironment = None) -> addnodes.pending_xr
     else:
         kwargs = {}
 
-    return pending_xref('', nodes.Text(text),
+    if env.config.python_smart_reference:
+        shortname = text.split('.')[-1]
```

Comment:
This is not good way for a nested class. But no way to get `__qualname__` because python domain should not import the python module dynamically.